### PR TITLE
Ensure we don't lose precision when exporting to txt

### PR DESCRIPTION
**Pull request by melhashash**
Set precision when exporting to COLMAP text files to avoid precision loss. (mostly when the scene is in global coordinates)

### CODE REVIEW

**Comment by pmoulon** [comment]
Thank you for this contribution.
Could you add a variable in the binary int `floating_point_precision_digit = 16;` and use it as a parameter for each function.
This way our user community could chose whatever value they would like to have easily.

**Comment by melhashash** [comment]
> 

Sure, a global variable is OK?

**Comment by melhashash** [comment]
@pmoulon updated by adding an argument option. 
If not set, the default is 16.
If set < 6 we set to 6 which is the default value used by `std` for precision if not set at all.

**Comment by pmoulon** [comment]
LGTM, I think your code editor is using tabs, and not 2spaces as 'tab'

File: src/software/SfM/export/main_openMVG2Colmap.cpp [hunk 1]
```diff
@@ -36,9 +36,13 @@ using namespace openMVG::features;
 
 bool CreateLineCameraFile(  const IndexT camera_id, 
                             std::shared_ptr<openMVG::cameras::IntrinsicBase> intrinsic,
-                            std::string & camera_linie)
+                            std::string & camera_linie,
```
[the remark this answers is not in the document]
Comment (melhashash) [reply src/software/SfM/export/main_openMVG2Colmap.cpp]: Fixed, spent half of the day adjusting indentation :)

**Comment by pmoulon** [comment]
Running Continous Integration build and unit test. But it LGTM
Thank you for your contribution and time to adjust the tabs ;-)

File: src/software/SfM/export/main_openMVG2Colmap.cpp [hunk 2]
```diff
@@ -435,8 +450,9 @@ int main( int argc , char ** argv )
     OPENMVG_LOG_INFO 
               << "Usage: " << argv[0] << '\n'
               << "[-i|--sfmdata] filename, the SfM_Data file to convert\n"
-              << "[-o|--outdir] path where cameras.txt, images.txt and points3D.txt will be saved";
-
+              << "[-o|--outdir] path where cameras.txt, images.txt and points3D.txt will be saved"
+              << "\n[Optional]\n"
+              << "[-p|--precision] sets the decimal precision to be used to format floating-point values (default = 16)";
```
[the remark this answers is not in the document]
Comment (melhashash) [reply src/software/SfM/export/main_openMVG2Colmap.cpp]: fixed

**Review by melhashash — COMMENTED** [review]
Showing the value of `floating_point_precision_digit` instead of hard coding it.

**Comment by melhashash** [comment]
> Running Continous Integration build and unit test. But it LGTM Thank you for your contribution and time to adjust the tabs ;-)

Sure, anytime! You are the one to thank for this awesome amount of work!

**Pull request merged**